### DISCUSSION TITLE
Why standard package management command result difference with WinGet command?

### DISCUSSION THREAD

**Discussion by soroshsabz** [microsoft/winget-cli#2955] [PowerShell]
ITNOA

As you can see below

```
PS C:\Users\sooro> Get-PackageProvider

Name                     Version          DynamicOptions
----                     -------          --------------
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, …
PowerShellGet            2.2.5.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, In…
WinGet                   0.0.8.0          AcceptLicense, Detailed
```

and 

```
PS C:\Users\sooro> Get-PackageSource

Name                             ProviderName     IsTrusted  Location
----                             ------------     ---------  --------
nuget.org                        NuGet            False      https://api.nuget.org/v3/index.json
PSGallery                        PowerShellGet    False      https://www.powershellgallery.com/api/v2
msstore                          WinGet           True       https://storeedgefd.dsx.mp.microsoft.com/v9.0
winget                           WinGet           True       https://cdn.winget.microsoft.com/cache
```

but when I try using standard package management commands, output of commands does not correct as you can see below

```
PS C:\Users\sooro> Find-Package firefox -Source winget
Find-Package: No match was found for the specified search criteria and package name 'firefox'. Try Get-PackageSource to see all available registered package sources.
```

but when I try this command with `Find-WinGetPackage` I see correct result

‍‍‍‍‍‍‍
```
PS C:\Users\sooro> Find-WinGetPackage firefox

ID                                Version         Source
--                                -------         ------
nulldev.ProfileSwitcherforFirefox 0.1.1           winget
Mozilla.Firefox.Nightly           108.2211.1408.0 winget
Mozilla.Firefox.ESR               102.7.0         winget
Mozilla.Firefox.DeveloperEdition  110.09          winget
Mozilla.Firefox.Beta              109.0           winget
Mozilla.Firefox                   109.0.1         winget
filips.FirefoxPWA                 2.4.0           winget
i2p.firefox                                       winget

```

How to resolve this problem?

**Comment by denelon** [comment]
We haven't published an official package provider for WinGet. 

I believe the one you are using is https://www.powershellgallery.com/packages/WinGet/0.0.8 by @ethanbergstrom at https://github.com/ethanbergstrom/WinGet.

**Reply by ethanbergstrom** [reply]
Thanks for the tag, @denelon. 

@soroshsabz the PackageManagement provider I made for winget is very opinionated because the PackageManagement framework is very particular about the concept of a 'package name' and winget is very...lose with its search terms, with a single search string qualifying on ID, name, moniker and tag. 

To remove any risk of ambiguity (and unexpected outcomes) when using the winget provider in a script or with PowerShell DSC, it will only search for packages [by exact package ID](https://github.com/ethanbergstrom/winget/blob/v0.0.8/src/private/Find-WinGetPackage.ps1#L58-L62). This is shown in the [project readme documentation](https://github.com/ethanbergstrom/winget/tree/v0.0.8#search-for-a-package), too. 

If you have further questions or suggestions, feel free to open an issue on the project site @denelon linked above.
